Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -569,6 +569,220 @@ class UpstreamIntegrationTests: TestBase {
         assertExpectedEvents(ignoreUnexpectedEvents: true)
     }
 
+    // Tests datastream config overrides
+    func testSendEvent_withValidConfigOverrides_receivesExpectedNetworkResponse() {
+        // Setup
+        // Note: test constructs should always be valid
+        let interactNetworkRequest = NetworkRequest(urlString: createURLWith(locationHint: edgeLocationHint), httpMethod: .post)!
+        // Setting expectation allows for both:
+        // 1. Validation that the network request was sent out
+        // 2. Waiting on a response for the specific network request (with timeout)
+        networkService.setExpectationForNetworkRequest(networkRequest: interactNetworkRequest, expectedCount: 1)
+        let configOverrides = ["com_adobe_experience_platform": [
+                                    "datasets": [
+                                        "event": [
+                                            "datasetId": "do_not_delete_dataset2_e2e_testing_edge_konductor_configoverride"
+                                        ]
+
+                                    ]
+                                ],
+                                "com_adobe_analytics": [
+                                    "reportSuites": [
+                                        "mobile5.e2e.rsid2"
+                                    ]
+                                ]]
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"], data: ["data": ["test": "data"]], datastreamConfigOverride: configOverrides)
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // Verify
+        // Network response assertions
+        let matchingResponses = networkService.getResponsesFor(networkRequest: interactNetworkRequest, timeout: 5)
+
+        XCTAssertEqual(1, matchingResponses.count)
+        XCTAssertEqual(200, matchingResponses.first?.responseCode)
+    }
+
+    func testSendEvent_withInvalidConfigOverrides_invalidData_receivesExpectedNetworkResponseError() {
```

Comment:
is the intention to test with both invalid config overrides and data or just testSendEvent_withInvalidConfigOverrides_receivesExpectedNetworkResponseError ?

Reply:
I wanted to shwo the type of invalidConfigOverride, like using dummyValues. I see it might be confusing with the data parameter that we take in. I will update the name.